src/PureScript/Backend/Optimizer/Semantics.purs

```diff
@@ -994,6 +1041,19 @@ evalExternFromImpl env@(Env e) qual (Tuple analysis impl) spine = case spine of
   _ ->
     Nothing
 
+evalExternRefFromImpl :: Env -> Qualified Ident -> Tuple BackendAnalysis ExternImpl -> BackendSemantics
+evalExternRefFromImpl env qual (Tuple _ impl) = case impl of
+  ExternExpr group (NeutralExpr expr) ->
+    eval (envForGroup env (EvalExtern qual) InlineRef group) expr
```

Comment:
It may be worth restricting this to not evaluating all terms, and maybe just literals, data, etc.